### PR TITLE
Nuclear: Remove Package

### PR DESCRIPTION
**Pull request by Lunarequest**
###### Description of changes
This PR will remove the nuclear music player from nixpkgs. 

Nuclear is made by a person who goes by the moniker nukeop or nitehawk. Nukeop has actively behaved in a xenophobic and transphobic manner and has called maintainers of other project slurs and refused to integrate project due to the maintainer being transgender. I am requesting that nuclear be removed from the nixpkgs as this goes against the spirit of an inclusive community.Since this is a very clear cut situation of the lead dev being openly against minorities, I believe the package should be removed.

This is a thread by a ex-user of nuclear about this developer: https://twitter.com/plsheadpats/status/1486975900021313536

Attached are two images of interactions that I in the past have had with nukeop while I maintained the flathub and AUR packages.

nukeop_anti_trans
againstBLM

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by MagicRB** [comment]
I am opposed to this PR. I do not support the actions of the the author of nucluar, but there must be separation between the artist and his art. If @Lunarequest has a *personal* issue with said author, they can talk it out with him or not use his projects, personally. Don't drag your personal disputes into packaging.

I do not like, nor support transphobia and related phobias, but I am of the opinion that software should not be treated different because of its author.

If this goes through, itll be hard for me and many others to recommend nixpkgs. Thats my 2 cents.

**Comment by maxnatt** [comment]
Well, that looks like a personal issue, you can simply ignore the mentioned person indeed, instead of trying to censor his work 🤷🏻‍♂️

**Comment by Lunarequest** [comment]
How about 
Xenophobia and transphobia are not a personal issue. By treating it as such implies there isn't an entire community effected by it and it sends the message we do not care about hateful developers. If this is the stance nix takes. I can not support nix any further

**Comment by siraben** [comment]
According to your links, the behavior by Nukeop did not seem to take place on any official (or managed) Nix-related communication channels. While I agree that the behavior is despicable, unless there is a specific *technical* reason why having this package in Nixpkgs increases maintenance burden (or if upstream is hostile), I cannot support this PR.

**Comment by BesterCoder** [comment]
This is clearly not a personal issue.

Like Lunarequest said, it's about making sure that we don't support bigoted developers and keeping the community welcoming for everyone.

If you accept the work of bigoted people, you are also accepting their opinions. Code doesn't exist in a vacuum.

> If this goes through, itll be hard for me and many others to recommend nixpkgs. Thats my 2 cents.

If NixOS supports bigots, it should be hard for most people to recommend nixpgks. That's my 2 cents.

**Comment by Lunarequest** [comment]
> According to your links, the behavior by Nukeop did not seem to take place on any official (or managed) Nix-related communication channels. While I agree that the behavior is despicable, unless there is a specific _technical_ reason why having this package in Nixpkgs increases maintenance burden (or if upstream is hostile), I cannot support this PR.

By not enforcing coc outside of offcial nix channels we are setting a dangrouse precident. Anyone can get away with harrasment as long as its outside offical channels. And this isn't the first time someone has abused rules like this. As shown by the interactions between randi harpper another frebsd dev.

Source for the gamer gate stuff: https://youtu.be/Ib7tFvw34DM?t=1909

**Comment by 7c6f434c** [comment]
> If you accept the work of bigoted people, you are also accepting their opinions. Code doesn't exist in a vacuum.

Just for the records, there have been cases where upstream authors came to Nix* spaces demanding we do not package their nominally opensource software, and in some cases the authors were effectively told to go away, and the packages stayed. I do not like the incentives we will get in that context…

**Comment by ethindp** [comment]
How exactly does this indicate that we are accepting the opinions of biggited developers by accepting and allowing people to use their code? There is a significant difference between the code and the developer. I'm sure we have some software in here somewhere that Richard Stallman wrote (ahem, GCC?) and yet we aren't removing any of that because of his later actions and the disputes/controversies surrounding him. I find it disturbing that people are willing to support the sensoring of certain individuals' code based off of interactions that they have had with that individual on platforms not managed by nixpkgs. It raises the question of "who's next"? Using this logical framework, I should be able to remove any package you submitt because of something you say to me on some other network or platform. I find this disturbing; its disturbing to see corporations firing people or taking actions against them for things they say on their (personal) Facebook or Twitter accounts, when they aren't representing that corporation on those accounts. I do not want to see something similar befall open source projects. If the developer was actively zenophobic and transphobic on a GitHub issue, PR, etc., I would be more open to the proposal. But what rules, exactly, has the author of this project broken? Did he come onto the NixOS forums and post zenophobic and transphobic content? Are we willing to risk trying to enforce the NixOS community code of conduct/rules on platforms that are not actually a part of the NixOS community at all? That I feel sets a dangerous precedent and is a very slippery-slope justification.

File: pkgs/top-level/all-packages.nix
Comment (dasJ) [inline pkgs/top-level/all-packages.nix:24468]: Technical hint - while I do not endorse this PR for reasons mentioned by others, it should still be correct if it should any chance of merging.
Please add a `throw` to aliases.nix that mentions the removal of the package.

**Review by dasJ — CHANGES_REQUESTED** [review]

**Comment by alyssais** [comment]
Just wanted to weigh in and say that as a member of a class affected by the undeniable biogtry presented here, I don't think we should drop the package for this reason.  The developer does not directly benefit from people using their software, while users do directly benefit from it being available for them to use.

Bigots are not welcome in the Nix community (and again, what you have shown here is definitely bigotry), but I don't agree that simply having a package present in the repositories is an invitation for the developers to participate in our community.  I would not support them if they tried to _actually_ participate in our community, even if they kept those opinions outside of official project spaces.

**Comment by BesterCoder** [comment]
> its disturbing to see corporations firing people or taking actions against them for things they say on their (personal) Facebook or Twitter accounts, when they aren't representing that corporation on those accounts

The point of not accepting hate speech is not to protect the brand. The point is to you know, to not accept hate speech. So it shouldn't be allowed on any platform

**Comment by ethindp** [comment]
> The point of not accepting hate speech is not to protect the brand. The point is to you know, to not accept hate speech. So it shouldn't be allowed on any platform

Show me proof that the developer of this software has posted or otherwise submitted hate speech within the NixOS community. I don't see how the posted images are evidence of an actual violation of the CoC. Are we going to remove all the software by RMS and ESR too?

**Comment by ethindp** [comment]
Because RMS and ESR were disagreeable individuals. There were some big, big controversies surrounding RMS, at least. So we should therefore remove their software too, using this logic.

**Comment by LunaTheFoxgirl** [comment]
Seeing as the predominant stance here is that code which is non-essential and is run by people who are hateful bigots is welcome, I would like to officially request that my projects Inochi2D, Inochi Creator and Inochi Session *NOT* be put up on nixpkgs.

**Comment by grahamc** [comment]
Hey @Lunarequest, thanks for this PR.

We don't typically evaluate the morals of authors when considering what to include in the package set. If we did, I think it would be very easy to find lots of examples of packaged software which would be also be extremely technically challenging to remove.

I do think it is a good idea to, as a user, consider the origins of software I'm using. I choose all the time to use less good software in exchange for supporting people, projects, and ideals I agree with. Not to be too blunt about it, but choosing open source software at all often involves this very tradeoff.

It is my inclination to not accept this PR on this basis.

I appreciate this is probably not the answer you were hoping for, and I am sorry. Those screenshots are despicable. The racism and xenophobia is awful. The rising tide of mistreatment, violence, and the erosion of trans rights is frightening. As a project, it is the Nix community's and the NixOS Foundation's goal to create a welcoming and safe environment.

As Alyssa said, and I fully stand behind this: bigots are not welcome in the Nix community. It is not acceptable to behave here as your screenshots depict.

So, that's my piece. It is really hard, as a project, to manage the tensions and needs of different users and groups and the complicated ways this interdependent web of people and software tie us together. It is my hope and goal that it continues to feel safe to participating in the project and in community, even with ugly corners of the package set.

**Comment by grahamc** [comment]
I do want to also address some of the discussion here.

Bigotry and xenophobia are not personal issues to be ignored. Bigotry and xenophobia are community issues. As a community we can't ask for that behavior and attitude to be checked at the door. Many people in our community are in marginalized groups and it is up to us to show them that we care for them, value their participation, and to provide an environment to allow for that. People in marginalized communities face adversity that many of us have never seen. It is important that we actively foster the community we want, and "show don't tell" our values.

I think it is also important to remember that open source is people, and that people don't easily fit into a logical framework. I hope that we can all do each other the favor of giving ourselves the grace to learn and grow together.

**Comment by adisbladis** [comment]
With what @grahamc said in https://github.com/NixOS/nixpkgs/pull/170283#issuecomment-1108923149 I think it's time to close this PR.

**Comment by marcan** [comment]
Have you considered the option of textually warning users who choose to install packages like this one about the views of upstream and how they conflict with those of the NixOS project itself? That would allow them to make an informed choice as to what packages to use based on their own personal judgement, as well as consider the option of using the software but staying away from upstream communities or support.

Just an idea.

**Comment by ethindp** [comment]
@marcan I don't really see how this would be feasible. Something like this is something that has too high of a chance of introducing bias towards an author or group of individuals. Not to mention no other package manager does this. I mean: I get rocking the boat and all, because change is good, but I'm not really sure how this would work without introducing a high risk of users just going "I don't like this author but I want this package in nixpkgs, so I'll just add this warning in here". It would be something we would need to verify, for every package that contains this warning, and that doesn't seem remotely practical given that nixpkgs has the largest collection of packages of any Linux distro (or so I'm told).

**Comment by marcan** [comment]
I'm thinking about a context similar to this PR, where upstream has shown actively harmful toxicity in their project spaces and someone has noticed and provided context here. It wouldn't be something you just throw in without supporting documentation. For example, the rule could be that such messages always have to have an associated GitHub issue/PR link where documentation was provided and any relevant discussion happened.

**Comment by lambdadog** [comment]
I can't help but back @marcan here as that being the *least* Nixpkgs should be doing and follow up with a further comment that @grahamc's point seems relatively null and void, for one specific reason:

> We don't typically evaluate the morals of authors when considering what to include in the package set. If we did, I think it would be very easy to find lots of examples of packaged software which would be also be extremely technically challenging to remove.

This package well and truly is *not* technically challenging to remove and this PR shows that. It begins to feel like an argument akin to "I can't solve racism so there's no point in fighting against it on a case by case basis."

While it may disrupt some users who use nuclear in their configuration, the old functional programming adage seems to apply here: "Avoid 'success at all costs'"

If the cost we're willing to pay for a minor success (avoiding inconveniencing a presumably small segment of users who include nuclear in their configuration) is endorsing discriminatory and toxic individuals, then NixOS and Nixpkgs have failed to be non-discriminatory as much as we want to cite that we don't allow discrimination in our own curated spaces.

If users wish they can certainly create their own repo akin to [guix-unfree](https://github.com/guix-users/guix-nonfree) but it does Nixpkgs a disservice to prioritize success over the well-being of their users and community members, which is the only way I can view the decision to not remove a trivially removable package.

For users who don't view packaging nuclear as an endorsement of an author's views, I can't help but wonder where that line would be drawn: If the package dev put "Installable with Nix!" and a link to NixOS.org right next to a manifesto for their hateful rhetoric in the project's README, would that be enough? Or would it still not be our concern?

**Comment by lambdadog** [comment]
Apologies, as I'm aware this has the potential to cause off-topic discussion about the "validity" of my statements, but:

I truly question what exactly anyone laugh-reacting to taking a stance against hate speech and specifically transphobic hate speech is trying to say when as we speak lawmakers are making literal torture legal (but only for trans people!) with so-called antiscientific "conversion therapy" and actively working to kill trans people, specifically *children*, by making life-saving medical treatment illegal.

👎 as much as you want to show disagreement, but laugh reacting to taking a stance against transphobia is frankly *incredibly* inappropriate given current events.

**Comment by grahamc** [comment]
@lambdadog Thanks for saying so re the laugh-react person. I've also noticed that behavior, and have been considering acting upon it. I'm happy to respond to your other comments as well, but am stepping away from dinner to issue a 30d ban against maxnatt, who has not otherwise engaged with this repository in any way.

**Comment by ethindp** [comment]
I'm honestly divided and unsure where my stance should be on this issue. I've pointed ot my thoughts on the Matrix room and in here -- that code, I think, should be separate from the author -- but as has been pointed out to me, on both here and within Matrix, code does not live in a vacuum. I am however uncomfortable applying this kind of decision on only certain packages without due consideration, particularly since I don't feel that applying it to, say, this package would be a fair application of the CoC. However, I do acknowledge that I may be too pacifistic here; I am just trying to figure out both my stance and what I believe would be a reasonable and just application of the CoC, in my opinion. I am but one member of the NixOS community, however, and if the community chooses to take action, I will stand with it, regardless of my own feelings or thoughts on this matter. I hope however that both sides can be considered equally before a final decision is reached.

**Comment by lambdadog** [comment]
@ethindp Thank you for being open minded and open about your thoughts!

I'm generally of the belief that this class of issues tends to be unclear for those who aren't personally affected by it. It's an unfortunate part of the human condition that we value our personal experiences far more highly than data and others' lived experiences and frankly, although it's uncomfortable to say, it's only my experience being discriminated against as a trans woman that's let me be as cognizant as I am of other groups' experiences with discrimination. 

It's been pointed out to me as I've talked about this on twitter that it's worthwhile thinking about this from a community perspective as opposed to "taking a hard stance against hate speech", which are pretty words but unfortunately abstract.

While it does signal that Nix is willing to "break bread" with hateful and discriminatory individuals for the sake of their software, more importantly, what happens when there are issues with Nuclear's packaging in nixpkgs and a maintainer is then required to work with the author to solve them? The obvious answer is "a non-trans person handles that communication", but I'd frankly hope that even if one wasn't trans they would take offense to this user's hate speech. At that point we're left with a community that, by necessity, includes those willing to turn a blind eye to hate speech as part and parcel of maintaining software written by said hate speech peddlers.

Personally, as a trans woman I find myself uncomfortable with the idea of sharing a community with anyone willing to "put up" with someone's hateful views against me for the sake of their entirely replaceable software.

As the software becomes less replaceable it becomes a more complicated issue but that's not the case with Nuclear so while it's worth considering in the future (it may be worth opening an issue on, although my condolences to the one who has to moderate it) it's not specifically relevant to this situation.

**Comment by ethindp** [comment]
@lambdadog I would definitely take offense if someone just said "let a non-trans person handle the communication between the author and maintainer"; such things sound like a (very non-obvious) form of discrimination. As someone who is blind and faces discrimination myself, I would be quite upset if any community did that to anybody, since communication should not be limited to someone with a particular set of characteristics or attributes (e.g. gender, sexual orientation, ...). Thank you for explaining your thoughts so clearly -- I believe I am beginning to understand how destructive an action like this could be, since it in effect almost requires that the community discriminate against its own members (!) and thereby violate its own code of conduct, which completely invalidates it and makes us all a bunch of hypocrites. I would like to hear more input though; I'm obviously no administrator of the community, but I would like to try to ensure that actions like this aren't taken too far, since things like this can easily be taken to extremes if we aren't careful.
As a proposal, I think it might be a wise choice to remove a package if and only if it can be determined that keeping the package within the repository would harm the community as a whole or members within, since keeping such a package would solicit communication between the maintainer(s) of the package in question and the author of that package, and the maintainer(s) may be impacted by keeping the package within the repository by being required to communicate with the author of the package should that be necessary. I know that the maintainer(s) could be altered, should the maintainer(s) not be willing to maintain the package any more, but I consider this to be an unusual action at best, and definitely not one I would be willing to undertake. I'm definitely open to other ideas and I'd like to be more informed about this though, since I feel that my knowledge on things like this is limited.

**Comment by grahamc** [comment]
As a note, I'm pretty introverted and spent 4h+ today on calls doing very mentally taxing social, extroverted stuff. Part of it was meeting with people to discuss this very PR. I'm mentally a puddle of mush. However, it was important to me to reply to some of the new discussion today. If something seems wack, I'd ask for the opportunity to clarify tomorrow after some sleep.

---

To start with: I care not one lick about the users of Nuclear or the convenience of having it, not having it, or having it and then removing it. This isn’t about success, or failure, or convenience.

I spent a few hours yesterday and today thinking through this on my own, with colleagues, community members, and members of other projects. Here are a few things that came out of it.

# Protecting Users

Early in the thread, lunarequest said that keeping it in the package set means allowing bigoted and hostile developers to be involved. To be clear: it does not. The author of this code, possessing those views, is not welcome to be part of this community. We are not “breaking bread” with them. We are not “putting up” with their behavior. If the Nuclear package breaks, nobody is required to fix it. 

I recognize that having their software packaged makes it possible to engage with their code, and could lead users of Nixpkgs to engage with the author.

## Warning users

At another point, marcan suggests marking packages like this one with a warning to indicate that the project’s values conflict with Nixpkgs. You could imagine something like the browser plugin “shinigami-eyes” which identifies and warns about it.

This is effectively adding a database of bigots to Nixpkgs. I have several reasons that this should not be part of Nixpkgs, but probably the most important one: imagine the contribution flow of adding a project or author to the list. It would be a nightmare, and I don’t think the project would or could survive. Furthermore, as a fellow community member points out: an immutable log you can never get out of is punitive for people who change.

Unfortunately, topics involving people are very difficult and sticky. Things are not always clear, and context is frequently missing or at least unevenly distributed.  

# Endorsement of the author

The project is specifically *not* endorsing the author’s bigotry.

Ultimately, decisions to include/exclude would mean implementing a subjective standard. It is a certainty that the subjective decisions don’t match essentially anyone’s position.

By choosing to remove some software, we are opting in to *automatically endorsing* all the remaining software in nixpkgs. There are a significant number of packages in the set that I cannot endorse.

# Comparing to other package sets

Most package sets in the world have a much stricter set of rules about what is included. Many of them are technical, or a bureaucratic process, or involve an in-person key signing with an existing maintainer.

A grim reality is, today, lots of important software from unpleasant people is packaged in other sets because their utility and importance is high. That software is “excusable” because we almost can’t imagine life without them. 

I believe the ease of contributing packages to Nixpkgs is one of its greatest strengths. However, it creates fertile ground to bring issues like this to the forefront by making the situation possible.

# Commenting is free

It is very easy for people who have very little history with the project to make grand statements and have it appear as though it is representative of the project. This is complicated by the ease by which someone can get a “Member” badge: anyone who maintains a package is a “member”, whether or not they can even merge a pull request or commit to the repository.

When I take a moderating action on GitHub or Matrix it is often because someone has raised a concern, pointing it out. Afterwards, I very often get several private messages saying thank you.

Unfortunately this means fewer people share their concerns publicly, and the project appears to have a weaker “immune response” than it does.

# The NixOS community is *pretty good*, and could do better

To start with: we need to do better. We need to be clearer about the project’s values and live them consistently.

Threads like this one do happen from time to time.

And, as a project in general, we work hard to eliminate and moderate out bigotry and unwelcoming behavior. Moderation is really hard, and everybody’s standard is slightly different.

We’re not the best, we have plenty of room to do better, but I firmly believe that the foundation’s leadership is committed to the values.

**Comment by lambdadog** [comment]
I appreciate your response and the labor you've put into this all, thank you for continuing to moderate on this difficult issue and I'd like to specifically note that nothing I say further here requires a rapid response.

-----

> ### Endorsement of the author
> The project is specifically not endorsing the author’s bigotry.
> 
> Ultimately, decisions to include/exclude would mean implementing a subjective standard. It is a certainty that the subjective decisions don’t match essentially anyone’s position.
> 
> By choosing to remove some software, we are opting in to automatically endorsing all the remaining software in nixpkgs. There are a significant number of packages in the set that I cannot endorse.

I fundamentally cannot agree that removing this package means endorsing all other packages in nixpkgs. There is a third option: A commitment to having discussions about packages with harmful authors when concerns are brought forward and taking action based on the results of said discussions.

It shouldn't be underplayed that this is a significant commitment, and I'm not going to pretend it's not. But as you say we need to be better, and FOSS communities in general are consistently behind the curve on this.

It's very easy to say the project isn't endorsing the author's bigotry, or even if it is, packaging it it nixpkgs isn't endorsing said bigotry, but at the end of the day those are just words. The author refuses to integrate with projects led by transgender people *because they are transgender*, is this not already the project reflecting the author's bigotry? Is use of the project, even exempting this case, not granting the author social capital to spew their hateful rhetoric, especially in open source communities where use of a piece of software often involves engaging with the author? Then in what way is nixpkgs not endorsing the use of a piece of software by *specifically enabling users to use it through official channels?*

> #### Warning users
> At another point, marcan suggests marking packages like this one with a warning to indicate that the project’s values conflict with Nixpkgs. You could imagine something like the browser plugin “shinigami-eyes” which identifies and warns about it.
> 
> This is effectively adding a database of bigots to Nixpkgs. I have several reasons that this should not be part of Nixpkgs, but probably the most important one: imagine the contribution flow of adding a project or author to the list. It would be a nightmare, and I don’t think the project would or could survive. Furthermore, as a fellow community member points out: an immutable log you can never get out of is punitive for people who change.
> 
> I'm reminded of the difficulty of applying technical solutions to a social problem.
> 
> Unfortunately, topics involving people are very difficult and sticky. Things are not always clear, and context is frequently missing or at least unevenly distributed

I think we both agree that it would be a messy and unideal solution to add a database of bigots to Nixpkgs, there's no contention on that, but we seem to disagree that this is the end of the discussion in any way. It's not something any functional programmer likes to admit given the field-wide obsession with elegant and correct solutions, but any experienced developer knows that there's a time to implement less than ideal solutions because a solution is required nonetheless.

Mind you, I'm not saying that a "database of bigots" is the right solution by far, I would (as I've argued) reach for simply removing packages on a case by case basis, but regardless of the final solution "a database of bigots in nixpkgs" being an unideal solution isn't the be-all-end-all.

People are difficult, which is another thing we agree on, but removing the package based on a discussion of the harmful behaviors of the author is not "solving a social problem with a technical solution", it's "solving a social problem with a social solution, which results in technical changes". There needs to be no commitment to being perfect, if the author improves as a person then the package can later be added back to Nixpkgs through the same ad-hoc methodology. All of these are a reflection of the issue-tracker centric methodology Nixpkgs uses for *all* user-contributed changes already.

No one is requiring that nixpkgs be perfect, simply that it *tries to be better*. People are a complicated problem, social solutions are often messy, *so aim for correctness in process rather than correctness in result*. If there is data that's missed in a discussion that results in a package being removed or not, then when that data shows up, reopen the discussion and amend the technical side based on new conclusions.

> I believe the ease of contributing packages to Nixpkgs is one of its greatest strengths. However, it creates fertile ground to bring issues like this to the forefront by making the situation possible.

If contributing packages to Nixpkgs is easy, then allow disputing their inclusion to be similarly easy. By keeping the solution ad-hoc you allow correctness in process while simultaneously allowing incorrect conclusions, with the ability to amend those decisions as new data is uncovered.

**Comment by f0x52** [comment]
> I find myself uncomfortable with the idea of sharing a community with anyone willing to "put up" with someone's hateful views against me for the sake of their entirely replaceable software.

I agree with @lambdadog, there's no need go over / endorse all other packages, as long as we remain committed to handle similar issues to this one as they come up for other packages. I think it's good to invite this sort of discussion and think about packages (maintainers) critically, even if it's a lot harder to moderate.

**Comment by ethindp** [comment]
I agree with @f0x52. Its good to have discussions like this before we make decisions that could potentially impact the community.

**Comment by Lunarequest** [comment]
Also could someone look into whats been marked off-topic. There are some on topic comments by @ethindp that have been marked offtopic.

**Comment by jkachmar** [comment]
> I believe the ease of contributing packages to Nixpkgs is one of its greatest strengths. However, it creates fertile ground to bring issues like this to the forefront by making the situation possible.

Sorry if I'm stepping outside of my lane by commenting here, but doesn't this sort of offer a neat solution to this problem?

Namely: if it's so easy for users to package things for inclusion within nixpkgs **and** layer their own packages on top of nixpkgs snapshots, then is it really so unreasonable to err on the side of removing software which is contentious within the community (according to the agreed-upon Code of Conduct that contributors would be held to, themselves)?

Anyone who is inconvenienced by this would be able to trivially include such a package by writing a derivation of their own.

**Comment by IreneKnapp** [comment]
Hey all. For those who don't know me, I'm a trans-feminine non-binary plural system with a long history of advocating for queer people and other people from marginalized backgrounds. While I haven't verified the allegations about Nuclear, I am disturbed by them and take the situation seriously. I'm keeping this post as short as I can, because I don't want to get into a situation where we're all writing lengthy essays at each other and nobody is able to respond in full... I think a finer-grained back-and-forth will be more productive.

I feel like this conversation is kind of intense, and I wanted to remind everyone that there are possible approaches to this very real problem, other than (1) accepting the PR, or (2) rejecting the PR and taking no other action. I do note (3) the suggestion to print a warning message, which I think from the social side of things has most of the same problems as (1) but certainly it's worth discussing.

I know from firsthand experience that NixOS is significantly better than most technical communities at inclusion. Things aren't perfect, but I think they're improving. I hope everyone can give each other the benefit of the doubt that we all want what's best for the community, including trans people within the community - we just have different types of experience that lead us to different conclusions about what that is.

I encourage any trans people on this thread who haven't previously met Graham to not treat him as an opponent. I have always known him to be a strong ally. Ally is a verb, it's about what people **do**, and Graham has done quite a bit.

I do think there is a very real capacity issue in terms of the social resources that are consumed by discussions of this type, and I don't think the Nix community is in a place where we have enough capacity for managing controversy to deal with a large number of discussions about the merits of package authors. I do think it is important to recognize that there is no real chance of this being a one-off. Whatever we do here will set an example for future removal requests, and therefore I think it is incredibly important not only to arrive at the correct answer, but to do so via a process that embodies justice. That is: If the decision is made by shouting, we are setting a binding precedent that shouting is how decisions get made. I think that would be very bad.

I'd like to ask, therefore, for people's ideas about how decisions like this **should** be made in the Nix community.

Okay - that was longer than I'd hoped, but I think it all needed to be said. I'll end there for now, but I hope to be available to participate in the thread going forward.

**Comment by 7c6f434c** [comment]
> more importantly, what happens when there are issues with Nuclear's packaging in nixpkgs and a maintainer is then required to work with the author to solve them?

As we kept packages where upstream authors came here and demanded we drop the package, inclusion in Nixpkgs _does not_ mean endorsement of possibility of communicating with the upstream author. We can be consciously aware that communication with upstream is not an option and carry on. 

Even support for legal sanctions against the upstream _for creating and distributing the software in the first place_ does not normally translate into demands for removal of a package from Nixpkgs.

**Comment by IreneKnapp** [comment]
Yeah, some sort of explicit separation like that seems pretty reasonable to me as an outcome. I would absolutely oppose Nuclear being depended on by stdenv (very unlikely, as stdenv is mostly compiler stuff), for all the reasons you've given, since I think that's the closest nixpkgs has to any concept of endorsement. Really the vast majority of nixpkgs is un-endorsed in my opinion, but this is not something the community has any sort of official position on, so there's room for disagreement.

I am, of course, not any sort of decision-maker here, I'm only chiming in to facilitate communication. Just to make sure that's clear.

**Comment by 7c6f434c** [comment]
> since I think that's the closest nixpkgs has to any concept of endorsement.

I guess «at least we actually expect those to work» could be somewhat stretched as endorsement of channel blocking jobs, but even there it is a stretch. The current state of an entire _Tier-2_ platform (Darwin aka macOS) is more of an uneasy truce than common agreement, and I think for every proposed approach to stdenv there someone keeps the opinion that this approach does not merit full technical endorsement.

> Currently as it stands nixpkgs is closer to the Archlinux Users Repository vs a distros packaging which for a person outside looking inside isn't clear. 

Well, if we take Debian as example, they have packages in `main` where communication with the upstream is clearly not an option and presumably they do not endorse the upstream author's views on Debian project.

Actually, if you consider «can communicate with upstream» a requirement for core dependencies, Nixpkgs is bad for technical reasons: it requires a bit of work to find out _who_ is even actually up to date about the question you are interested in. It might be the person who is not the last contributor, the last merger, nor the listed maintainer of the package. This is even coming from the same reasons as Nixpkgs being effectively unable to even have global positions on most questions.

**Pull request closed without merging**
